frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -866,7 +865,8 @@ function ReaderHighlight:updateHighlight(index, side, direction, move_by_char)
     local new_end = highlight.pos1
     local new_text = self.ui.document:getTextFromXPointers(new_beginning, new_end)
     highlight.text = cleanupSelectedText(new_text)
-    self.ui:handleEvent(Event:new("AnnotationsModified", { highlight, bookmark_before }))
+    local highlight_before = { pos0 = highlight_beginning, pos1 = highlight_end }
+    self.ui:handleEvent(Event:new("AnnotationsModified", { highlight, highlight_before }))
```

Comment:
(This is usual between us two :))
Not fond of "downgrading" a highglight to only the fields our code _currently_ use: you're breaking the contract of the array having annotation items, and the day we add a feature that needs more fields, we'll have to revisit this and wonder why we did this downgrade, and if it is risky replacing it with a proper item.